allensdk/brain_observatory/behavior/behavior_project_cache/tables/util/metadata_parsers.py

```diff
@@ -0,0 +1,147 @@
+import pandas as pd
+import numpy as np
+from typing import Optional
+
+#####################################################################
+#
+#                   VBO PROJECT CONSTANTS
+#
+#####################################################################
+
+NUM_STRUCTURES_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 2,
+    "VisualBehaviorMultiscope4areasx2d": 4,
+}
+
+NUM_DEPTHS_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 4,
+    "VisualBehaviorMultiscope4areasx2d": 2,
+}
+
+
+############################
+#
+#     from session type
+#
+############################
+
+
+def parse_behavior_context(session_type: str) -> Optional[str]:
+    """Return if session is passive or active.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    behavior_type : str
+        Return string describing active or passive session.
+    """
+    is_passive = (
+        'passive' in session_type
+        and "OPHYS" in session_type
+    )
+    if is_passive:
+        behavior_type = "passive_viewing"
+    else:
+        behavior_type = "active_behavior"
+    return behavior_type
+
+
+def parse_stimulus_set(session_type: str) -> Optional[str]:
+    """Return the name of the image set or gratings.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    stim_set : str
+        Name of stimulus set shown for session.
+    """
+    session_type_split = session_type.split("_")
+    stimulus_type = session_type_split[2]
+    if stimulus_type == "images":
+        image_set_letter = session_type_split[3]
+        stim_set = f"{stimulus_type}_{image_set_letter}"
+    elif stimulus_type == "gratings":
+        stim_set = "gratings"
+    else:
+        raise ValueError(f"Session_type {session_type} not formatted as "
+                         "expected.")
+    return stim_set
+
+
+def add_experience_level(
```

Comment:
sorry to be a pain, but it looks like this doesn't really fit with the rest of the metadata parsers. Nothing is being parsed. Can this be moved to `prior_exposure_processing.py` ?